cluster/gke/config-test.sh

```diff
@@ -18,11 +18,22 @@
 CLUSTER_NAME="${CLUSTER_NAME:-${USER}-gke-e2e}"
 NETWORK=${KUBE_GKE_NETWORK:-e2e}
 NODE_TAG="k8s-${CLUSTER_NAME}-node"
-IMAGE_TYPE="${KUBE_GKE_IMAGE_TYPE:-container_vm}"
+IMAGE_TYPE="${KUBE_GKE_IMAGE_TYPE:-cos}"
 ENABLE_KUBERNETES_ALPHA="${KUBE_GKE_ENABLE_KUBERNETES_ALPHA:-}"
 
 KUBE_DELETE_NETWORK=${KUBE_DELETE_NETWORK:-true}
 
+# Set e2e test context through NODE_OS_DISTRIBUTION.
+case $(kubectl get node -o=custom-columns=IMAGE:.status.nodeInfo.osImage | grep -v -m 1 IMAGE) in
+*"Container-Optimized OS from Google")
+	NODE_OS_DISTRIBUTION="gci"
+	;;
+*"Debian GNU/Linux 7 (wheezy)")
+	NODE_OS_DISTRIBUTION="debian"
+	;;
+esac
+echo "Setting NODE_OS_DISTRIBUTION=$NODE_OS_DISTRIBUTION"
```

Comment:
this should probably be `${NODE_OS_DISTRIBUTION-}` instead of `$NODE_OS_DISTRIBUTION`, since it will be unset when `cluster/gke/config-test.sh` is sourced efore the cluster is started (or anytime else there is no existing cluster).